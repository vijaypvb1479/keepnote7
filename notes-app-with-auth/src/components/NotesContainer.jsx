import React from 'react';
import { Grid } from '@material-ui/core';
import Note from './Note';

const NotesContainer = (props) => (
    <Grid container spacing={8}>
        {
            props.notes.map(note =>  {
                return (
                    <Grid key={note.id} item xs={12} sm={4} xl={3}>
                        <Note note={note} handleRemoveNote={props.handleRemoveNote} />
                    </Grid>
                );
            })
        }
    </Grid>
);

export default NotesContainer;