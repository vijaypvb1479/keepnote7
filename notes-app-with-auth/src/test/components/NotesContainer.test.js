import React from 'react';
import { mount } from 'enzyme';
import NotesContainer from '../../components/NotesContainer';
import { MemoryRouter } from 'react-router-dom';
import { createSerializer } from 'enzyme-to-json';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

let notes, handleRemoveNote, wrapper;

beforeEach(() => {
    handleRemoveNote = jest.fn();
    notes = [
        {
            "noteTitle": "note-1",
            "noteDescription": "note-1 description",
            "id": 2
        }, {
            "noteTitle": "note-2",
            "noteDescription": "note-2 description",
            "id": 1
        }];

    wrapper = mount(
        <MemoryRouter>
            <NotesContainer
                notes={notes}
                handleRemoveNote={handleRemoveNote} />
        </MemoryRouter>
    );

})

test('test Grid View ', () => {

    expect(wrapper).toMatchSnapshot();

});