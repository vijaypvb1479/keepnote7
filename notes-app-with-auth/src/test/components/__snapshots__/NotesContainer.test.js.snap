// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Grid View  1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-8-23"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-4-47 MuiGrid-grid-xl-3-88"
  >
    <div
      className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99"
    >
      <div
        className="MuiCardHeader-root-127"
      >
        <div
          className="MuiCardHeader-content-130"
        >
          <span
            className="MuiTypography-root-133 MuiTypography-headline-138 MuiCardHeader-title-131"
          >
            note-1
          </span>
        </div>
        <div
          className="MuiCardHeader-action-129"
        >
          <a
            href="/edit-note/2"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root-174 MuiIconButton-root-168"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-173"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-177"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-190"
              />
            </button>
          </a>
        </div>
      </div>
      <div
        className="MuiCardContent-root-186"
      >
        note-1 description
      </div>
      <div
        className="MuiCardActions-root-187 Note-deleteIcon-98"
      >
        <button
          className="MuiButtonBase-root-174 MuiIconButton-root-168 MuiCardActions-action-189"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-173"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-177"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-190"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-4-47 MuiGrid-grid-xl-3-88"
  >
    <div
      className="MuiPaper-root-100 MuiPaper-elevation1-103 MuiPaper-rounded-101 MuiCard-root-99"
    >
      <div
        className="MuiCardHeader-root-127"
      >
        <div
          className="MuiCardHeader-content-130"
        >
          <span
            className="MuiTypography-root-133 MuiTypography-headline-138 MuiCardHeader-title-131"
          >
            note-2
          </span>
        </div>
        <div
          className="MuiCardHeader-action-129"
        >
          <a
            href="/edit-note/1"
            onClick={[Function]}
          >
            <button
              className="MuiButtonBase-root-174 MuiIconButton-root-168"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="0"
              type="button"
            >
              <span
                className="MuiIconButton-label-173"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-177"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                </svg>
              </span>
              <span
                className="MuiTouchRipple-root-190"
              />
            </button>
          </a>
        </div>
      </div>
      <div
        className="MuiCardContent-root-186"
      >
        note-2 description
      </div>
      <div
        className="MuiCardActions-root-187 Note-deleteIcon-98"
      >
        <button
          className="MuiButtonBase-root-174 MuiIconButton-root-168 MuiCardActions-action-189"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-173"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-177"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-190"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
